fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultScheduledDateGenerator.java

```diff
@@ -96,10 +96,37 @@ public LocalDate generateNextRepaymentDate(final LocalDate lastRepaymentDate, fi
                         loanApplicationTerms.getNumberOfdays());
             }
         }
-        
+        if (currentCalendar == null && holidayDetailDTO.getWorkingDays().getExtendTermForRepaymentsOnHolidays()) {
```

Comment:
This sounds like a design decision that I will need to ask about. But I agree that something needs to change. Imagine the case where a loan has been dispersed, and then a holiday is added that affects a future repayment. This patch does not account for this case, and I'm guessing that it should. Good catch.